84 wpla 2x bug precipitation always zero with knmi uurgegevens

rflinnenbank opened the pull request:
### Fix for issue #84

The number of stored decimals for float values has been doubled from 2 to 4. This fixes a bug where the field "Uurgegevens" would always end up as 0 for text based files, due to the default harmonization level being in **m/h** rather than **mm/h**. Due to API consistency this choice of unit will remain the same for version 2.x of the API calls. 

**_It is not done to change the output of an API without proper warning in advance,  which is why we change the unit depth, rather than the unit itself..._**

rflinnenbank committed:
[2022-12-07 20:31]: Fixed bug #84

FIXED:
- Text based files would round to 2 decimals while the precipitation for uurgegevens was expressed in m/h, resulting in almost always values rounding to zero.

NOTES:
- The fix handles the rounding and not the base type, as that has been as it was since version 1 of the API. The third API call version will have mm/h as the default unit for precipitation.

Signed-off-by: Raoul Linnenbank <58594297+rflinnenbank@users.noreply.github.com>

johandebraak reviewed (approved):
Wat kleine suggesties en vragen, maar over het algemeen top!

rflinnenbank commented on pyproject.toml:
```diff
@@ -11,32 +11,33 @@ include = [
 pytest-cov = "^4.0.0"
 isort = "^5.10.1"
 black = "^22.10.0"
+jupyter = "^1.0.0"
```
"Jupyter" is part of the development requirements _(please note the [tool.poetry.group.dev.dependencies] dependency group)_. Why this is a part of the dev utilities, is to more easily test and validate the proper working certain structures and access methods during development.

rflinnenbank commented on weather_provider_api/routers/weather/sources/knmi/client/knmi_downloader.py:
```diff
@@ -58,23 +59,22 @@ def knmi_download_request(self):
+            self.download_folder.mkdir()
         else:
-            for root, dirs, files in os.walk(file_storage_directory):
+            for root, dirs, files in os.walk(self.download_folder):
```
Very nicely noticed, but in this format "dirs" should've probably been used to remove the files. I didn't write this particular segment of code myself, but it seems as if to properly remove the files we'd need the root, sub-directory and the files. (so all of the variables)

I'll look through this and alter the code accordingly.

rflinnenbank replied:
And adjusted

rflinnenbank commented on weather_provider_api/routers/weather/sources/knmi/models/harmonie_arome.py:
```diff
@@ -4,42 +4,40 @@
+        # Pre-work
         super().__init__()
-        self.id = "arome"
+        self.id = 'arome'
```
The title holds the Dutch _(or even possibly the French)_ word "Harmonie". We stick with the original name for purposes of identifying the original source. 

As for the 'id' value, we try to keep that at a single word to keep the API access as easy as possible. If we were to implement a second  Harmonie Arome version from the KNMI (there actually is a second version based on a rotated grid), we'd likely rename both to 'arome_py1' and 'arome_py3' for a clear reference. _(py1 and py3 are version specific denominators used by KNMI themselves)_

rflinnenbank commented on weather_provider_api/routers/weather/sources/knmi/models/harmonie_arome.py:
```diff
@@ -58,63 +57,57 @@ def __init__(self):
             return list(self.to_si.keys())
 
-        new_factors = []
+        validated_factors = []
```
A very valid argument. The **_request_weather_factors()** function is unfortunately the very definition of code debt. It can easily be changed, but originally it has been added to each and every module, and it has proven to be a pain to transfer to a singular location. 

Changing it here, would mean having to change it for every model. This may happen in the future as a part of the EOL cycle of version 2 of this project, but version 3 currently takes precedence to this.

rflinnenbank commented on weather_provider_api/routers/weather/sources/knmi/models/harmonie_arome.py:
```diff
@@ -58,63 +57,57 @@ def __init__(self):
-            f_up = f.upper()
-            f_low = f.lower()
+        for factor in factor_list:
+            factor_lowercase = factor.lower()
```
Not in this version of the **_request_weather_factors()** function, because somewhere along the line some of the code probably was cleaned to up better match the actual usage for this model. 

Very much unfortunately, as with the previous remark, this function is pretty much the heaviest form of code debt that this  project has and while during the EOL cycle this probably will be cleaned up, until at least version 3 is live, this code will probably remain.

rflinnenbank committed:
[2022-12-12 10:18] Multi-location requests are working

CHANGES:
 - Changed the comment for the first day of the repository to correspond to the actual moment in the value.
 - Altered the way the download folder would be cleaned up before usage.
Signed-off-by: Raoul Linnenbank <58594297+rflinnenbank@users.noreply.github.com>

rflinnenbank committed:
Merge branch 'release-2.31' into 84-wpla-2x-bug-precipitation-always-zero-with-knmi-uurgegevens
